spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java

```diff
@@ -631,6 +651,137 @@ public void setFailFast(boolean failFast) {
 
 	}
 
+	/**
+	 * High (and some medium) priority Streams properties and a general properties bucket.
+	 */
+	public static class Streams {
+
+		private final Ssl ssl = new Ssl();
+
+		/**
+		 * Kafka streams application.id property; default spring.application.name.
+		 */
+		private String applicationId = "${spring.application.name}";
```

Comment:
Is that supposed to be SpEL? I'd prefer if we perform that detection elsewhere (in the caller with access to the `Environment`.

Reply:
I used a property placeholder which gets resolved as expected in the test case; I can move it to the caller.

Reply:
We don't do property placeholder in properties. Yes please.